src/MSBuildBinLogQuery/Graph/ProjectNode_BeforeThis.cs

```diff
@@ -0,0 +1,27 @@
+using System.Collections.Generic;
+using Microsoft.Build.Logging.Query.Component;
+
+namespace Microsoft.Build.Logging.Query.Graph
+{
+    public class ProjectNode_BeforeThis :
```

Comment:
I'm not sure I understand why the before/after relationships require specific types. Can you explain?

Reply:
Discussed offline.